libs/vkd3d/heap.c

```diff
@@ -80,7 +85,10 @@ static ULONG STDMETHODCALLTYPE d3d12_heap_Release(d3d12_heap_iface *iface)
     TRACE("%p decreasing refcount to %u.\n", heap, refcount);
 
     if (!refcount)
-        d3d12_heap_destroy(heap);
+    {
+        d3d12_heap_decref(heap);
+        d3d12_device_release(heap->device);
```

Comment:
Bug to access heap->device after d3d12_heap_decref.

Reply:
fixed